pyfixmsg/reference.py

```diff
@@ -277,11 +277,18 @@ def __init__(self, element, spec):
         assert element.tag == "message"
         self.msgtype = element.get('msgtype')
         self.name = element.get('name')
-        self.composition = _extract_composition(element, spec)
+        self.composition = self._extract_composition_including_header_trailer(element, spec)
         self.groups = {group.count_tag.tag: group for group in _get_groups(self.composition)}
         self.sorting_key = _extract_sorting_key(self.composition)
         self._spec = spec
 
+    def _extract_composition_including_header_trailer(self, element, spec):
+        """ Add standard FIX header and trailer around element composition. """
+        returned = _extract_composition(spec.tree.find('header'), spec)
+        returned = returned + _extract_composition(element, spec)
+        returned = returned + _extract_composition(spec.tree.find('trailer'), spec)
+        return returned
+
```

Comment:
Did you run your code with Fix spec 4.2, Does it work with Fix spec 5.0 ? Since the 'header' and 'trailer' sections have no tags in spec 5.0, so we should make sure that the changes works for both version 4 and 5.

Reply:
Sorry for the delay, just got back to this.
I do not think it relates to this patch as FIX5.0 fails with exact same stack trace even without this patch applied. Probably a separate issue?
Thank you for helping.
(I am surprised the QuickFix spec for FIX5.0 has empty header and trailer tags.)

Reply:
francoisverbeek has made a PR for that tag ordering issue and it has been merged. maybe it is also helpful for your case.